tracer/src/Datadog.Trace/DuckTyping/ILHelpersExtensions.cs

```diff
@@ -341,6 +350,40 @@ internal static void WriteTypeConversion(this LazyILGenerator il, Type actualTyp
             }
         }
 
+        internal static void CheckTypeConversion(Type actualType, Type expectedType)
```

Comment:
nit: I think we should write a big warning both here and in `WriteTypeConversion()` that the checks in both these methods have to match - if you change either, you need to change both

Reply:
Comment added in https://github.com/DataDog/dd-trace-dotnet/pull/2716/commits/7a8a8942d10547d8c3d564471e01f9631f2c635d